### ISSUE TITLE
Bug: no-unused-vars ignoreRestSiblings option

### ISSUE THREAD

**Issue by mike-robertson** [eslint/eslint#8170]
I've put the ignoreRestSiblings option to true, but function signatures don't seem to be included in this option.
>  Using a Rest Property it is possible to “omit” properties from an object, but by default the sibling properties are marked as “unused”. With this option enabled the rest property’s siblings are ignored.

I'm essentially using it like this for this purpose. If I spread `buttonProps` onto my Tag, and that tag happens to be a plain html type, react will complain since it doesn't understand the `sheet` var (injected by a library HOC). Here's the code:
```js
const Button = ({ onClick, classes, className, tag, children, sheet, ...buttonProps }) => {
  const Tag = tag || 'button';

  return (
    <Tag
      {...buttonProps}
      onClick={onClick}
      className={classnames(classes.container, className)}
    >{children}</Tag>
  );
};
```
Ideally, this ignoreRestSiblings option would apply to functions which use destructuring as well.

**Comment by eslintbot** [comment]
Hi @mike-robertson, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by not-an-aardvark** [comment]
I think this is a duplicate of https://github.com/eslint/eslint/issues/8119, which has been fixed on master but hasn't been in a release yet.

**Comment by mike-robertson** [comment]
Yep, looks like it. I should have searched the issues!
